log/log.go

```diff
@@ -0,0 +1,14 @@
+package log
+
+// Logger is the least-common-denominator interface for all log operations.
+type Logger interface {
+	With(Field) Logger
+	Logf(string, ...interface{}) error
```

Comment:
In my opinion, the `f` (format) bit of `Logf` is redundant by the ability to set fields with `With`.  

Usually, the form 
```go
Logf(msg, value1, value2)
``` 
is an anti-pattern that should be replaced by 
```go
With({key1: value1},{key2: value2}).Log(msg)
```

By having `Logf` and `With`, there's now two ways to do the same thing (embedding values in the log), one of which is less useful because it is not easily machine-readable.